tljh/installer.py

```diff
@@ -242,21 +242,42 @@ def ensure_user_environment(user_requirements_txt_file):
                 )
                 to_upgrade.append(pkg)
 
-        cf_pkgs_to_upgrade = list(set(to_upgrade) & {"conda", "mamba"})
-        if cf_pkgs_to_upgrade:
-            conda.ensure_conda_packages(
-                USER_ENV_PREFIX,
-                # we _could_ explicitly pin Python here,
-                # but conda already does this by default
-                cf_pkgs_to_upgrade,
-            )
-        pypi_pkgs_to_upgrade = list(set(to_upgrade) & {"pip"})
-        if pypi_pkgs_to_upgrade:
-            conda.ensure_pip_packages(
-                USER_ENV_PREFIX,
-                pypi_pkgs_to_upgrade,
-                upgrade=True,
-            )
+    # force reinstall conda/mamba to ensure a basically consistent env
+    # avoids issues with RemoveError: 'requests' is a dependency of conda
+    # only do this for 'old' conda versions known to have a problem
+    # we don't know how old, but we know 4.10 is affected and 23.1 is not
+    if not is_fresh_install and V(package_versions.get("conda", "0")) < V("23.1"):
```

Comment:
I don't really like what I ended up doing here, but I bracketed the force reinstall on a version _older_ than what's going to be in 1.0, with the hope that we actually remove this step when we drop support for upgrading from earlier than 1.0.

I think you need pretty old conda to get here, but that's what we have in 0.2. It really is a special-case to handle:

- quite old conda
- inconsistent environment in conda's own dependencies (caused by our own `pip install --upgrade` commands)

but this is what you'll get if you start with tljh 0.2 and haven't upgraded conda, so we _should_ handle it. I think it is appropriate to treat it specially, as if users have broken their own envs such that conda itself doesn't work, I think it's appropriate for us to not handle that and require users to fix their conda (or delete their user env to start fresh) before upgrading. But since _we_ are creating a broken environment, we need to handle it, at least for the very specific case of upgrading from a previous tljh known to have this issue.